test/test_models.py

```diff
@@ -7,17 +7,29 @@
 import unittest
 import traceback
 import random
+import inspect
 
 
-def set_rng_seed(seed):
+EPSILON = 1e-5  # small value for approximate comparisons/assertions
```

Comment:
This is only used once, maybe delete

Reply:
It started off being more broadly used, but I refactored most of that away. I still favor keeping a named constant here - either `EPSILON` or `TOLERANCE` - rather than just inserting a magic number with no explanation.